docs/built-in-functions.rst

```diff
@@ -664,6 +664,22 @@ Math
         >>> ExampleContract.foo(9.0)
         3.0
 
+.. py:function:: isqrt(a: uint256) -> uint256
+
+    Return the square root of the provided decimal number, using `Solmate's square root algorithm <https://github.com/transmissions11/solmate/blob/9cf1428245074e39090dceacb0c28b1f684f584c/src/utils/FixedPointMathLib.sol#L166>`_.
```

Comment:
probably the best place for this attribution is in the comments or commit message. having it in the documentation might lead to an expectation that the semantics or performance are identical to solmate, which we don't want to imply. in the documentation itself we might want to say what the general method is (babylonian right?) and any semantics that are important (i think we do want to mention that it rounds down)